pkg/resource/logging-secret/alloy-logging-secret.go

```diff
@@ -1,13 +1,69 @@
 package loggingsecret
 
 import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"text/template"
+
 	v1 "k8s.io/api/core/v1"
 
+	"github.com/Masterminds/sprig/v3"
+
+	"github.com/giantswarm/logging-operator/pkg/common"
 	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	loggingcredentials "github.com/giantswarm/logging-operator/pkg/resource/logging-credentials"
+)
+
+const (
+	AlloyLokiURLEnvVarName           = "LOKI_URL"
+	AlloyTenantIDEnvVarName          = "TENANT_ID"
+	AlloyBasicAuthUsernameEnvVarName = "BASIC_AUTH_USERNAME"
+	AlloyBasicAuthPasswordEnvVarName = "BASIC_AUTH_PASSWORD" // #nosec G101
 )
 
+var (
+	//go:embed alloy/logging-secret.yaml.template
+	alloySecret         string
+	alloySecretTemplate *template.Template
+)
+
+func init() {
+	alloySecretTemplate = template.Must(template.New("logging-secret.yaml").Funcs(sprig.FuncMap()).Parse(alloySecret))
+}
+
 func GenerateAlloyLoggingSecret(lc loggedcluster.Interface, credentialsSecret *v1.Secret, lokiURL string) (map[string][]byte, error) {
```

Comment:
Are you not missing a call to this function in the controller?

Reply:
Nevermind

Reply:
This function is already being called here https://github.com/giantswarm/logging-operator/blob/070de870fd41dcaff7e1903406c4b115e32c937b/pkg/resource/logging-secret/logging-secret.go#L30

Reply:
Yes sorry I had to dig deeper :)